spring-integration-redis/src/test/java/org/springframework/integration/redis/util/RedisLockRegistryTests.java

```diff
@@ -922,6 +951,14 @@ private void waitForExpire(String key) throws Exception {
 		assertThat(n < 100).as(key + " key did not expire").isTrue();
 	}
 
+	private void waitForExpire2(String key) throws Exception {
+		StringRedisTemplate template = createTemplate();
+		int n = 0;
+		while (n++ < 50 && template.keys(this.registryKey + ":" + key).size() > 0) {
+			Thread.sleep(100);
+		}
```

Comment:
Doesn't look like this new method has some value, unless it introduces unnecessary thread blocking, including Redis call.

I believe in your second new test we could just use existing `waitForExpire()`.
However, at the same time your `testRenewalExceptionOnExpire()` is fully similar to existing `testExceptionOnExpire()`.
So, that one is redundant.

You might think about adding a new test for the new `renewLock()` API and model it after  `JdbcLockRegistryTests.testLockRenew()` and `testLockRenewLockNotOwned()`, respectively.